### PR TITLE
Final component configurations

### PR DESCRIPTION
**Pull request by erenbesel**

### CODE REVIEW

File: AdyenCard/Components/BCMC/BCMCComponent.swift [hunk 1]
```diff
@@ -12,11 +12,14 @@ import UIKit
...
+    /// Initializes the BCMC Component.
+    /// - Parameters:
+    ///   - paymentMethod: BCMC payment method.
+    ///   - apiContext: The API context.
+    ///   - configuration: The configuration of the component.
     public init(paymentMethod: BCMCPaymentMethod,
-                configuration: CardComponent.Configuration = CardComponent.Configuration(),
                 apiContext: APIContext,
-                style: FormComponentStyle = FormComponentStyle()) {
+                configuration: CardComponent.Configuration = CardComponent.Configuration()) {
```
Comment (mohammedDehairy) [inline AdyenCard/Components/BCMC/BCMCComponent.swift]: maybe this is more compact :-)

```
configuration: CardComponent.Configuration = .init()
```

File: AdyenDropIn/Models/DropInConfiguration.swift [hunk 2]
```diff
@@ -55,51 +55,107 @@ public extension DropInComponent {
...
                     allowPreselectedPaymentView: Bool = true) {
             self.apiContext = apiContext
+            self.style = style
             self.allowsSkippingPaymentList = allowsSkippingPaymentList
             self.allowPreselectedPaymentView = allowPreselectedPaymentView
         }
 
     }
+    
+    struct ActionComponentConfiguration {
```
Comment (mohammedDehairy) [inline AdyenDropIn/Models/DropInConfiguration.swift:78]: maybe `ActionComponent` is enough? I know its my code, but I realized its a bit much 😅

File: Tests/AdyenTests/Adyen Tests/Utilities/ComponentManagerTests.swift [hunk 3]
```diff
@@ -120,15 +116,14 @@ class ComponentManagerTests: XCTestCase {
...
                                    order: nil,
                                    presentationDelegate: presentationDelegate)
         
         XCTAssertEqual(sut.storedComponents.count, 4)
         XCTAssertEqual(sut.regularComponents.count, numberOfExpectedRegularComponents)
         
-        XCTAssertEqual(sut.storedComponents.compactMap { $0 as? Localizable }.filter { $0.localizationParameters == config.localizationParameters }.count, 4)
-        XCTAssertEqual(sut.regularComponents.compactMap { $0 as? Localizable }.filter { $0.localizationParameters == config.localizationParameters }.count, 3)
+        XCTAssertEqual(sut.storedComponents.compactMap { $0 as? Localizable }.filter { $0.localizationParameters == config.localizationParameters }.count, 2)
+        XCTAssertEqual(sut.regularComponents.compactMap { $0 as? Localizable }.filter { $0.localizationParameters == config.localizationParameters }.count, 0)
```
Comment (mohammedDehairy) [inline Tests/AdyenTests/Adyen Tests/Utilities/ComponentManagerTests.swift]: this is not needed any more, right?

File: AdyenCard/Components/BCMC/BCMCComponent.swift (hunk 1 above)
Comment (erenbesel) [reply AdyenCard/Components/BCMC/BCMCComponent.swift]: Conflicted with `ActionComponent` in a file they were both accessed so this looks ok :)

File: AdyenDropIn/Models/DropInConfiguration.swift (hunk 2 above)
Comment (erenbesel) [reply AdyenDropIn/Models/DropInConfiguration.swift:78]: Conflicted with ActionComponent in a file they were both accessed so this looks ok :)

**Review by nauaros — APPROVED** [review]
nice!

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**